Update Time Series Insights spec with GA features, including Operations.

sandshadow opened the pull request:
Highlighted changes:
1. Add a GA spec
2. Fix the inheritance problems with EventSourceResource and its properties and parameters. This was based on the changes @mcardosos recently made, but in a way that kept the discriminator on the EventSourceCreateOrUpdateParameters.
3. Add new GA features, including operations, to both spec.
Changes have been tested by generating a CSharp client using the latest version of autorest.

There are no autorest linter errors.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
Update Time Series Insights spec with GA features, including Operations.
Highlighted changes:
1. Add a GA spec
2. Fix the inheritance problems with EventSourceResource and its properties and parameters. This was based on the changes @mcardosos recently made, but in a way that kept the discriminator on the EventSourceCreateOrUpdateParameters.
3. Add new GA features, including operations, to both spec.
Changes have been tested by generating a CSharp client using the latest version of autorest.

There are no autorest linter errors.

olydis commented:
@ravbhatnagar new features

olydis commented on specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/Swagger/timeseriesinsights.json:
```diff
@@ -34,14 +34,42 @@
     }
```
please don't change the folder structure! Note that this PR introduces a folder `Swagger` which also requires changing all the example paths - please adjust the structure to be consistent with the repo's guidelines. 🙂

sandshadow replied:
Reverted. For some reason I thought the swagger subfolder was required.

olydis commented on specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/Swagger/timeseriesinsights.json:
```diff
@@ -1192,6 +1275,11 @@
+        "storageLimitExceededBehavior": {
+          "type": "string",
+          "description": "The behavior the Time Series Insights service should take when the environment's capacity has been exceeded. If \"PauseIngress\" is specified, new events will not be read from the event source. If \"PurgeOldData\" is specified, new events will continue to be read and old events will be deleted from the environment. The default behavior is PurgeOldData.",
+          "enum": [ "PurgeOldData", "PauseIngress" ]
```
note that without `x-ms-enum`, this will not be modeled as an enum in generated SDKs - please add `x-ms-enum` (see other enums in this spec)

olydis commented on specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/Swagger/timeseriesinsights.json:
```diff
@@ -1253,7 +1342,8 @@
       "required": [
-        "kind"
+        "kind",
+        "properties"
```
Invalid definition: Where is this property called `properties`?

sandshadow replied:
You're right. It's the derived classes (EventHubEventSourceResource, IoTHubEventSourceResource) that have the properties collections.

olydis commented on specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/Swagger/timeseriesinsights.json:
```diff
@@ -1364,7 +1456,8 @@
       "required": [
-        "kind"
+        "kind",
+        "properties"
```
same as above, there is no `properties` here - if a schema further up the inheritance chain has that property, it must be `required` up there

sandshadow replied:
Removed, thanks.

Commit:
PR feedback:
1) return to previous repo structure
2) Add x-ms-enum to storageLimitExceededBehavior
3) Remove required "properties" envelope from objects that don't have properties. Move to derived object if necessary.

olydis commented:
Sweet, thanks for the changes! The [linter](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/300401717#L738) has some remaining errors (but as the bot shows, you've gotten rid of one, awesome!). Since you didn't introduce them with this PR, that's fine, just as always, raising awareness 😉

ravbhatnagar commented:
Looks good! signing off

AutorestCI commented:
No modification for AutorestCI/azure-sdk-for-node
